### ISSUE TITLE
HOOBS 3 not Supported on ARM 6L

### ISSUE THREAD

**Issue by vladiny** [hoobs-org/HOOBS#285]
Hello HOOBS,

Today I downloaded HOOBS v3.1.1 and followed the instructions to work on my raspberry pi. I was able to get through all the steps, but when I go to http://hoobs.local it stays on the screen below:
![Annotation 2019-12-05 171212](https://user-images.githubusercontent.com/58573890/70278501-67a0cc00-1782-11ea-9f94-664e4a15d38f.png)

Any suggestions?
I am able to SSH into the Raspberry Pi, but don't know the username or password. Thanks in advance!

**Comment by k1ul3ss** [comment]
I am on Fedora running 3.1.1 and having the same issue.

**Comment by philjones** [comment]
Same here - I installed the SD Card version from the image (using BalenaEtcher) onto a Raspberry Pi 1B.

Each of the requests just ends up with 502 Bad Gateway

![Screen Shot 2019-12-05 at 9 24 37 PM](https://user-images.githubusercontent.com/692676/70290379-b744bf00-17a5-11ea-8ca9-d5260c11db97.png)

**Comment by philjones** [comment]
Figured out how to SSH into the raspberry pi - looking at the syslog

```
Dec  5 21:41:40 hoobs systemd[1]: Started HOOBS.
Dec  5 21:41:40 hoobs systemd[1]: hoobs.service: Main process exited, code=killed, status=4/ILL
Dec  5 21:41:40 hoobs systemd[1]: hoobs.service: Failed with result 'signal'.
Dec  5 21:41:43 hoobs systemd[1]: hoobs.service: Service RestartSec=3s expired, scheduling restart.
Dec  5 21:41:43 hoobs systemd[1]: hoobs.service: Scheduled restart job, restart counter is at 35.
Dec  5 21:41:43 hoobs systemd[1]: Stopped HOOBS.
```

**Comment by philjones** [comment]
from #284, saw a suggestion to check the node version

```
hoobs@hoobs:/var/log $ node -v
Illegal instruction
```

**Comment by philjones** [comment]
hoobs@hoobs:/var/log $ cat /proc/cpuinfo
processor	: 0
model name	: ARMv6-compatible processor rev 7 (v6l)
BogoMIPS	: 697.95
Features	: half thumb fastmult vfp edsp java tls 
CPU implementer	: 0x41
CPU architecture: 7
CPU variant	: 0x0
CPU part	: 0xb76
CPU revision	: 7

Hardware	: BCM2835
Revision	: 000f
Serial		: 000000006f82890c
Model		: Raspberry Pi Model B Rev 2
hoobs@hoobs:/var/log $ sudo apt update
Hit:1 http://archive.raspberrypi.org/debian buster InRelease
Hit:2 http://raspbian.raspberrypi.org/raspbian buster InRelease                                                              
Hit:3 https://dl.yarnpkg.com/debian stable InRelease                                                                         
Reading package lists... Done                          
Building dependency tree       
Reading state information... Done
All packages are up to date.
```
hoobs@hoobs:/var/log $ whereis node
node: /usr/bin/node /usr/local/bin/node /usr/share/man/man1/node.1.gz
hoobs@hoobs:/var/log $ /usr/bin/node -v
v10.15.2
hoobs@hoobs:/var/log $ /usr/local/bin/node -v
Illegal instruction
hoobs@hoobs:/var/log $ apt list nodejs
Listing... Done
nodejs/stable,now 10.15.2~dfsg-2+rpi1 armhf [installed,automatic]
hoobs@hoobs:/var/log $ 
```

It doesn't look like there is an official build of nodejs of 12.x
https://nodejs.org/download/release/latest-v12.x/

https://github.com/nodejs/build/issues/1677#issuecomment-486204349

**Comment by philjones** [comment]
Since node 12 isn't supported on the raspberry pi and I've already gone this far with HOOBS3 I will document my progress to see how far I get...

```
hoobs@hoobs:~ $ wget https://unofficial-builds.nodejs.org/download/release/v12.13.1/node-v12.13.1-linux-armv6l.tar.gz
hoobs@hoobs:~ $ tar -xf node-v12.13.1-linux-armv6l.tar.gz --directory /usr/local --strip-components 1
hoobs@hoobs:~ $ node --version
v12.13.1
```

checking on how HOOBS is doing now that there is a node version that runs
```
hoobs@hoobs:~ $ tail -f /var/log/syslog
Dec  5 22:13:27 hoobs systemd[1]: Stopped HOOBS.
Dec  5 22:13:27 hoobs systemd[1]: Started HOOBS.
Dec  5 22:13:27 hoobs systemd[1]: hoobs.service: Main process exited, code=killed, status=4/ILL
Dec  5 22:13:27 hoobs systemd[1]: hoobs.service: Failed with result 'signal'.
Dec  5 22:13:30 hoobs systemd[1]: hoobs.service: Service RestartSec=3s expired, scheduling restart.
Dec  5 22:13:30 hoobs systemd[1]: hoobs.service: Scheduled restart job, restart counter is at 620.
Dec  5 22:13:30 hoobs systemd[1]: Stopped HOOBS.
Dec  5 22:13:31 hoobs systemd[1]: Started HOOBS.
Dec  5 22:14:49 hoobs hoobs[1397]: - Plugins
Dec  5 22:14:49 hoobs hoobs[1397]: - Dependencies
Dec  5 22:17:01 hoobs CRON[1422]: (root) CMD (   cd / && run-parts --report /etc/cron.hourly)
Dec  5 22:17:07 hoobs hoobs[1397]: - Application
Dec  5 22:17:28 hoobs hoobs[1397]: innerError { Error: Cannot find module '../build/Debug/pty.node'
Dec  5 22:17:28 hoobs hoobs[1397]:     at Function.Module._resolveFilename (internal/modules/cjs/loader.js:581:15)
Dec  5 22:17:28 hoobs hoobs[1397]:     at Function.Module._load (internal/modules/cjs/loader.js:507:25)
Dec  5 22:17:28 hoobs hoobs[1397]:     at Module.require (internal/modules/cjs/loader.js:637:17)
Dec  5 22:17:28 hoobs hoobs[1397]:     at require (internal/modules/cjs/helpers.js:22:18)
Dec  5 22:17:28 hoobs hoobs[1397]:     at Object.<anonymous> (/home/hoobs/.hoobs/node_modules/node-pty/lib/unixTerminal.js:30:15)
Dec  5 22:17:28 hoobs hoobs[1397]:     at Module._compile (internal/modules/cjs/loader.js:689:30)
Dec  5 22:17:28 hoobs hoobs[1397]:     at Object.Module._extensions..js (internal/modules/cjs/loader.js:700:10)
Dec  5 22:17:28 hoobs hoobs[1397]:     at Module.load (internal/modules/cjs/loader.js:599:32)
Dec  5 22:17:28 hoobs hoobs[1397]:     at tryModuleLoad (internal/modules/cjs/loader.js:538:12)
Dec  5 22:17:28 hoobs hoobs[1397]:     at Function.Module._load (internal/modules/cjs/loader.js:530:3) code: 'MODULE_NOT_FOUND' }
Dec  5 22:17:28 hoobs hoobs[1397]: (node:1397) UnhandledPromiseRejectionWarning: Error: The module '/home/hoobs/.hoobs/node_modules/node-pty/build/Release/pty.node'
Dec  5 22:17:28 hoobs hoobs[1397]: was compiled against a different Node.js version using
Dec  5 22:17:28 hoobs hoobs[1397]: NODE_MODULE_VERSION 72. This version of Node.js requires
Dec  5 22:17:28 hoobs hoobs[1397]: NODE_MODULE_VERSION 64. Please try re-compiling or re-installing
Dec  5 22:17:28 hoobs hoobs[1397]: the module (for instance, using `npm rebuild` or `npm install`).
Dec  5 22:17:28 hoobs hoobs[1397]:     at Object.Module._extensions..node (internal/modules/cjs/loader.js:718:18)
Dec  5 22:17:28 hoobs hoobs[1397]:     at Module.load (internal/modules/cjs/loader.js:599:32)
Dec  5 22:17:28 hoobs hoobs[1397]:     at tryModuleLoad (internal/modules/cjs/loader.js:538:12)
Dec  5 22:17:28 hoobs hoobs[1397]:     at Function.Module._load (internal/modules/cjs/loader.js:530:3)
Dec  5 22:17:28 hoobs hoobs[1397]:     at Module.require (internal/modules/cjs/loader.js:637:17)
Dec  5 22:17:28 hoobs hoobs[1397]:     at require (internal/modules/cjs/helpers.js:22:18)
Dec  5 22:17:28 hoobs hoobs[1397]:     at Object.<anonymous> (/home/hoobs/.hoobs/node_modules/node-pty/lib/unixTerminal.js:26:11)
Dec  5 22:17:28 hoobs hoobs[1397]:     at Module._compile (internal/modules/cjs/loader.js:689:30)
Dec  5 22:17:28 hoobs hoobs[1397]:     at Object.Module._extensions..js (internal/modules/cjs/loader.js:700:10)
Dec  5 22:17:28 hoobs hoobs[1397]:     at Module.load (internal/modules/cjs/loader.js:599:32)
Dec  5 22:17:28 hoobs hoobs[1397]: (node:1397) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
Dec  5 22:17:28 hoobs hoobs[1397]: (node:1397) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
Dec  5 22:17:28 hoobs systemd[1]: hoobs.service: Succeeded.
```

**Comment by philjones** [comment]
Didn't have anything on hoobs.local so decided to try a reboot:

![Screen Shot 2019-12-05 at 10 29 14 PM](https://user-images.githubusercontent.com/692676/70293590-ba907880-17ae-11ea-843b-ab6cbe1a60d2.png)

**Comment by mkellsy** [comment]
HOOBS 3 I'd built for arm7l. Our image and install script uses arm7l, and is only available for RPI 3 3+ and 4.

Otherwise you will need to do a manual install.

**Comment by BobbySlope** [comment]
https://github.com/hoobs-org/HOOBS/wiki/5.1-Certified-Hardware

**Comment by k1ul3ss** [comment]
Why did you close this issue? It had nothing to do with hardware.

**Comment by mkellsy** [comment]
What device are you on?
When you run `uname -m` what is the output?
When you run `node -v` what is the output?

What I think is happening is the ARM chip you are running is the old 6l version. Node dropped support for arm6l.

If this is the case. You can try to use the method philjones uses, but I don't know if that works because HOOBS and see node-pty which requires a node-gyp build on install.

Because of this HOOBS 3 is officially only supported on the RPI 3B, 3B+ and 4B.

We have this documented here.
https://github.com/hoobs-org/HOOBS/wiki/5.1-Certified-Hardware

**Comment by vladiny** [comment]
Thanks philjones, that worked for me also. I was able to find the default username and password through google search:
username: hoobs
password: hoobsadmin
Being that it is not supported on a Raspberry Pi 1B because of ARM 6L, I will install on my Raspberry Pi 3B. 

mkellsy,

Thanks for the input. You should fix hoobs.org to state this as it is confusing and leads to believe HOOBS 3 will work on Raspbery Pi 1B. I would prefer for the Docker version so I can install in my synology server. Hopefully it will be released soon. Thanks for all your hard work.

**Comment by mkellsy** [comment]
I have it in our wiki
https://github.com/hoobs-org/HOOBS/wiki/5.1-Certified-Hardware

But I'm having the guy that maintains our website add this to the product pages.
"Compatible with Raspberry Pi 3B, 3B+ and 4B"

I'm the GitHub guy, I stay away from the website :)

**Comment by vladiny** [comment]
Thanks mkellsy!

I understand you have it in the wiki, but the website is incorrect, people purchasing your product may think it's compatible. I am glad you will have the person maintaining the website update with the correct information. Any update on the Docker release?

**Comment by mkellsy** [comment]
That issue is here
https://github.com/hoobs-org/HOOBS/issues/68

I have to wait until release issues are solved, before I start on that. (After Christmas). HOOBS is only 4 people, and I am the only developer. Sorry for the delay.

**Comment by aaronraimist** [comment]
I ran into the same problem and just wanted to say that I found the wording on https://github.com/hoobs-org/HOOBS/wiki/5.1-Certified-Hardware to be a bit confusing. I read:

> HOOBS releases are tested on this hardware

to mean exactly that, HOOBS 3 has been tested on Pi 3 and 4 and hasn't been tested on other Pis. My reading of this didn't lead me to believe it was **known** to be incompatible on the Pi Zero W.
